File: system-base/src/main/java/com/levnovikov/system_base/base_di/ComponentKey.java
package com.levnovikov.system_base.base_di;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import dagger.MapKey;

/**
 * Created by lev.novikov
 * Date: 2/12/17.
 */

@MapKey
@Target({ElementType.METHOD}) @Retention(RetentionPolicy.RUNTIME)
public @interface ComponentKey {
    Class<? extends ComponentBuilder> value();
}

File: system-base/src/main/java/com/levnovikov/system_base/base_di/ComponentKey.kt
package com.levnovikov.system_base.base_di

import dagger.MapKey
import kotlin.reflect.KClass

/**
 * Created by lev.novikov
 * Date: 2/12/17.
 */

@MapKey
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class ComponentKey(val value: KClass<out ComponentBuilder>)
